src/Scrypt/ScryptEncoder.cs

```diff
@@ -41,18 +41,22 @@ public class ScryptEncoder
         /// <summary>
         /// Size of salt in bytes.
         /// </summary>
-        private const int SaltLength = 32;
+        public static readonly int SaltLength = 32;
+
         private const int DefaultIterationCount = 16384;
         private const int DefaultBlockSize = 8;
         private const int DefaultThreadCount = 1;
 
         private readonly int _iterationCount;
         private readonly int _blockSize;
         private readonly int _threadCount;
+        private readonly byte[] _salt;
```

Comment:
Why is `_salt` a class attribute? This will make `Encode` not thread-safe anymore.

Reply:
Hi @viniciuschiele! Thank you for the comment. The `Encode` method wasn't thread safe to begin with because `RandomNumberGenerator.GetBytes` is not thread safe. A general practice most follow is to make the caller responsible for resource locking for thread safety related to individual methods, unless the class itself is running multiple threads internally or using overlapping callbacks of some kind. The exception to this rule is usually for methods marked `static`, which should ensure their own thread safety. Let me know if you have any other feedback. Thanks!

Reply:
Hi, based on [this](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.rngcryptoserviceprovider?view=netframework-4.8), `RNGCryptoServiceProvider` is thread-safe and `RandomNumberGenerator.Create()` also returns a `RNGCryptoServiceProvider` by default based on [that](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.cryptoconfig?redirectedfrom=MSDN&view=netcore-3.1).
My concern is that this change can potentially break existing apps, that's why I think this shouldn't be part of this PR.

Reply:
Thank you for the clarification. After some research based on your comment, I see the issue now. I agree on the point of it breaking existing apps. Even though this varies by implementation, it shouldn't cause a behavioral change that breaks existing consumers of the library who use older versions of .NET, since you still support them. I'll go ahead and remove that change and return it to the original with a new commit tomorrow.

I did want to point out that in .NET Core and .NET Framework, the implementation now just invokes [Windows API's BCryptGenRandom on Windows](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RandomNumberGeneratorImplementation.Windows.cs) and [OpenSSL's GetRandomBytes on Unix](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RandomNumberGeneratorImplementation.Unix.cs), so new consumers should not rely on thread safety of this method in _modern_ codebases.

I really appreciate your thoughtful response and for exercising care around breaking your users. Thank you again and you can expect the commit tomorrow.

Reply:
Thanks for providing the links to the source code, for me both implementations seem thread-safe and I couldn't find anything to say otherwise.

I agree that this may vary based on the implementation, I wonder if it would be better to instantiate a new `RandomNumberGenerator` every time the `Encode` is called.